aderyn_driver/src/driver.rs

```diff
@@ -205,3 +205,204 @@ fn detect_framework(path: PathBuf) -> Option<Framework> {
+            content: Arc::new(admin_contract_file_content),
+        })
+        .unwrap();
+
+        let solc = Solc::find_or_install_svm_version(format!("{}", version));
+        assert!(solc.is_ok());
+
+        println!("Solc binary: {:?}", solc.unwrap());
+    }
+
+    #[test]
+    fn can_generate_ast_for_admin_contract() {
+        let cargo_root = PathBuf::from(std::env::var("CARGO_MANIFEST_DIR").unwrap());
+        let tests_contract_playground_path = cargo_root
+            .join("../tests/contract-playground/")
+            .canonicalize()
+            .unwrap();
+
+        let admin_contract_file = tests_contract_playground_path.join("src/AdminContract.sol");
+        let admin_contract_file_content = std::fs::read_to_string(&admin_contract_file).unwrap();
+
+        // Step 1 - Gather the data to input to the compiler
+        let compiler_input = CompilerInput::new(
+            // This will work with root directory as well (it will fetch list of files)
+            admin_contract_file.as_path(),
+        )
+        .unwrap();
+
+        let ac_compiler_input = compiler_input.get(0).unwrap();
+
+        // Step 2 - Detect the version of solc that can be used with compiler input
+        let version = Solc::detect_version(&Source {
+            // This will install the compiler if not there
+            content: Arc::new(admin_contract_file_content),
+        })
+        .unwrap();
+
+        // Step 3 -  Get a representation of binary to use for compiling
+        let solc = Solc::find_or_install_svm_version(format!("{}", version)).unwrap();
+        let solc_bin = solc.solc.to_str().unwrap();
+        println!("Path to binary {}", solc_bin);
+        assert!(Path::new(solc_bin).exists());
+
+        // Step 4 - Run `solc --ast-compact-json <FILENAME.sol>`
+        let file_arg = ac_compiler_input
+            .sources
+            .first_key_value()
+            .unwrap()
+            .0
+            .to_str()
+            .unwrap();
+        let command = Command::new(solc_bin)
+            .args(["--ast-compact-json", file_arg])
+            .current_dir(tests_contract_playground_path)
+            .stdout(Stdio::piped())
+            .output();
+        //TODO: expect command to work here
+
+        if let Ok(command) = command {
+            // assert!(command.status.success()); // TODO: Investigate why it fails in CI
+            let stdout = String::from_utf8(command.stdout).unwrap();
+
+            println!("AST {}", stdout);
+            println!("If you are seeing this in CI, likely the above AST is empty because stdout of is streamed to github infra.")
+            // assert!(!stdout.is_empty());
+        }
+    }
+
+    #[test]
+    fn can_load_context_for_admin_contract() {
+        let cargo_root = PathBuf::from(std::env::var("CARGO_MANIFEST_DIR").unwrap());
+        let tests_contract_playground_path = cargo_root
+            .join("../tests/contract-playground/")
+            .canonicalize()
```

Comment:
@alexroan My goal for this PR is pretty scoped down. I don't want to make the whole aderyn project independent of foundry/hardhat JUST YET.

I want to first test this out with our `load_contract()` function which is what we use when we write tests for our detectors.

So I mean `load_contract('/blah/../src/path/to/solidity/file.sol')` should automatically download the relevant solc for that file, create ast and prepare the context loader. (caveat: solc would be spawned from `src` directory. Because We don't have any configuration yet, let's stick to defaults)

Maybe a bit of observation on how it performs (gather feedback, etc), and then after that let's tackle the beast! What do you say?

Reply:
Update: I'll use `load_contract_experimental` because I'm not even sure how to get these tests to pass in CI lol. I'm looking for some help in that aspect too :)